### PR TITLE
feat: implement epoch sync done

### LINKED ISSUE #738: Contracts: Implement epoch change done calls

### PR DESCRIPTION
**Pull request by karlwuest**
closes #738

### CODE REVIEW

File: contracts/walrus/sources/staking/staking_inner.move [hunk 1]
```diff
@@ -411,6 +415,30 @@ public(package) fun advance_epoch(self: &mut StakingInnerV1, mut rewards: Balanc
...
+    self: &mut StakingInnerV1,
+    cap: &mut StorageNodeCap,
+    clock: &Clock,
+) {
+    // Make sure the node hasn't attested yet, and set the new epoch as the last sync done epoch.
+    assert!(cap.last_epoch_sync_done() < self.epoch, EDuplicateSyncDone);
+    cap.set_last_epoch_sync_done(self.epoch);
+
+    let node_shards = self.committee.shards(&cap.node_id());
+    match (self.epoch_state) {EpochState::EpochChangeSync(weight) => {
```
Comment (damirka) [inline contracts/walrus/sources/staking/staking_inner.move]: Is the formatter giving this result? 😢

Comment (karlwuest) [reply contracts/walrus/sources/staking/staking_inner.move]: Let me check, I might have reverted all formatter changes after it broke the arguments list on the `register` function.

Comment (karlwuest) [reply contracts/walrus/sources/staking/staking_inner.move]: Ah wait, that wasn't even in this file. Yes, that's the formatter output 😬

Comment (karlwuest) [reply contracts/walrus/sources/staking/staking_inner.move]: I made it a bit nicer manually.

**Pull request merged**